CMakeLists.txt

```diff
@@ -29,7 +29,12 @@ function(boost_iostreams_option name description package version found target) #
 
     find_package(${package} ${version} REQUIRED)
     target_sources(boost_iostreams PRIVATE ${ARGN})
-    target_link_libraries(boost_iostreams PRIVATE ${target})
+
+    if(TARGET ${name})
```

Comment:
This logic does actually seem to be working as intended.
Inclined to withdraw this PR, but will revisit.